### PR TITLE
Implement space templates feature

### PR DESCRIPTION
**Pull request by wkhessairi**

### CODE REVIEW

File: integration/src/main/java/org/exoplatform/task/integration/SpaceTaskAppInstaller.java [hunk 1]
```diff
@@ -22,9 +22,10 @@
 import org.exoplatform.container.xml.ValueParam;
```
Comment (boubaker) [inline integration/src/main/java/org/exoplatform/task/integration/SpaceTaskAppInstaller.java]: We should completely remove this service

File: integration/src/main/resources/conf/portal/configuration.xml [hunk 2]
```diff
@@ -3,10 +3,6 @@
 <configuration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.exoplatform.org/xml/ns/kernel_1_2.xsd http://www.exoplatform.org/xml/ns/kernel_1_2.xsd"
                xmlns="http://www.exoplatform.org/xml/ns/kernel_1_2.xsd">
 
-  <component profiles="all">
```
Comment (boubaker) [inline integration/src/main/resources/conf/portal/configuration.xml]: Please move this file configuration inside the WAR extension to make it possible to override it.

And move the configuration that is actually defined in `task-management.war/WEB-INF/conf/configuration.xml` to a different configuration file to make it possible to override it as well.

File: integration/src/main/resources/conf/portal/configuration.xml [hunk 3]
```diff
@@ -142,5 +138,1084 @@
     </component-plugin>
   </external-component-plugins>
 
+  <external-component-plugins>
+    <target-component>org.exoplatform.social.core.space.spi.SpaceTemplateService</target-component>
+    <!-- Community space template -->
+    <component-plugin>
+      <name>Space Template Configuration</name>
+      <set-method>registerSpaceTemplatePlugin</set-method>
```
Comment (boubaker) [inline integration/src/main/resources/conf/portal/configuration.xml]: We should use the new method `extendSpaceTemplatePlugin` to inject task application inside an existing space template with an appropriate order without having to redefine the whole Space template.

File: integration/src/main/resources/conf/portal/configuration.xml [hunk 4]
```diff
@@ -142,5 +138,1084 @@
...
+              </collection>
+            </field>
+          </object>
+        </object-param>
+      </init-params>
+    </component-plugin>
+    <!-- end Team space template -->
+  </external-component-plugins>
+
   <import>jar:/conf/portal/task-notification-plugin.xml</import>
```
Comment (boubaker) [inline integration/src/main/resources/conf/portal/configuration.xml]: `<import>jar:/conf/portal/task-notification-plugin.xml</import>` has to be moved to the extension WAR as well to make it possible to override it.

File: integration/src/main/java/org/exoplatform/task/integration/SpaceTaskAppInstaller.java (hunk 1 above)
Comment (wkhessairi) [reply integration/src/main/java/org/exoplatform/task/integration/SpaceTaskAppInstaller.java]: done

File: integration/src/main/resources/conf/portal/configuration.xml (hunk 3 above)
Comment (wkhessairi) [reply integration/src/main/resources/conf/portal/configuration.xml]: done

File: integration/src/main/resources/conf/portal/configuration.xml (hunk 4 above)
Comment (wkhessairi) [reply integration/src/main/resources/conf/portal/configuration.xml]: ok

File: integration/src/main/resources/conf/portal/configuration.xml (hunk 2 above)
Comment (wkhessairi) [reply integration/src/main/resources/conf/portal/configuration.xml]: done

File: integration/src/main/resources/conf/portal/configuration.xml [hunk 5]
```diff
@@ -3,144 +3,5 @@
 <configuration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.exoplatform.org/xml/ns/kernel_1_2.xsd http://www.exoplatform.org/xml/ns/kernel_1_2.xsd"
```
Comment (boubaker) [inline integration/src/main/resources/conf/portal/configuration.xml]: This whole file has to be deleted and please add import statement inside task-management.war/WEB-INF/conf/configuration.xml

Comment (wkhessairi) [reply integration/src/main/resources/conf/portal/configuration.xml]: ok

File: task-management/src/main/webapp/WEB-INF/conf/task-addon/social-configuration.xml [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+              </collection>
+            </field>
+          </object>
+        </object-param>
+      </init-params>
+    </component-plugin>
+    <!-- end Team space template -->
+  </external-component-plugins>
+
+  <import>war:/conf/task-addon/notification/task-notification-plugin.xml</import>
```
Comment (boubaker) [inline task-management/src/main/webapp/WEB-INF/conf/task-addon/social-configuration.xml]: Move this to `WEB-IN/conf/configuration.xml`

Comment (wkhessairi) [reply task-management/src/main/webapp/WEB-INF/conf/task-addon/social-configuration.xml]: yes

**Pull request merged**